pkg/client/cache/eventqueue.go

```diff
@@ -400,6 +397,7 @@ func (eq *EventQueue) Resync() error {
 	for _, id := range eq.store.ListKeys() {
 		if !inQueue.Has(id) {
 			eq.queue = append(eq.queue, id)
+			eq.events[id] = watch.Modified
```

Comment:
Consider adding a comment as to why it's desirable to set the state to modified on resync.